src/app/drive/services/downloadManager.service.ts

```diff
@@ -0,0 +1,417 @@
+        ? {
+            credentials: {
+              user: workspaceCredentials.credentials.networkUser,
+              pass: workspaceCredentials.credentials.networkPass,
+            },
+            workspaceId: selectedWorkspace.workspace.id,
+            mnemonic: selectedWorkspace.workspaceUser.key,
+          }
+        : undefined;
+    return credentials;
+  };
+
+  readonly generateTasksForItem = async (downloadItem: DownloadItem): Promise<DownloadTask | undefined> => {
+    const itemsPayload = downloadItem.payload;
+    if (itemsPayload.length === 0) return;
+
+    const uploadFolderAbortController = new AbortController();
+    const abort = () => Promise.resolve(uploadFolderAbortController.abort('Download cancelled'));
+
+    const formattedDate = date.format(new Date(), 'YYYY-MM-DD_HHmmss');
+    let downloadName = `Internxt (${formattedDate})`;
+    if (itemsPayload.length === 1) {
+      const item = itemsPayload[0];
+      if (itemsPayload[0].isFolder) {
+        downloadName = item.name;
+      } else {
+        downloadName = item.type ? `${item.name}.${item.type}` : item.name;
+      }
+    }
+
+    let taskId = downloadItem.taskId;
+    if (taskId) {
+      tasksService.updateTask({
+        taskId,
+        merge: {
+          status: TaskStatus.Decrypting,
+          cancellable: true,
+          stop: abort,
+        },
+      });
+    } else if (itemsPayload.length > 1) {
+      taskId = tasksService.create<DownloadFilesTask>({
+        action: TaskType.DownloadFile,
+        file: {
+          name: downloadName,
+          type: 'zip',
+          items: downloadItem.payload as DriveItemData[],
+        },
+        showNotification: true,
+        cancellable: true,
+        stop: abort,
+      });
+    } else {
+      const item = itemsPayload[0];
+      if (item.isFolder) {
+        taskId = tasksService.create<DownloadFolderTask>({
+          action: TaskType.DownloadFolder,
+          folder: item as DriveFolderData,
+          compressionFormat: 'zip',
+          showNotification: true,
+          cancellable: true,
+          stop: abort,
+        });
+      } else {
+        taskId = tasksService.create<DownloadFileTask>({
+          action: TaskType.DownloadFile,
+          file: item as DriveFileData,
+          showNotification: true,
+          cancellable: true,
+          stop: abort,
+        });
+      }
+    }
+
+    const user = localStorageService.getUser();
+    if (!user) throw new Error('User not found');
+
+    const userCredentials: DownloadCredentials = {
+      credentials: {
+        user: user.bridgeUser,
+        pass: user.userId,
+      },
+      mnemonic: user.mnemonic,
+    };
+    const workspaceCredentials = this.getDownloadCredentialsFromWorkspace(
+      downloadItem.selectedWorkspace,
+      downloadItem.workspaceCredentials,
+    );
+    const credentials = downloadItem.downloadCredentials ?? workspaceCredentials ?? userCredentials;
```

Comment:
Maybe abstract this to a function as `getCredentialsByPriority` or something similar would be great to understand that this is for a specific reason